packages/core/src/router/router.ts

```diff
@@ -42,24 +39,21 @@ export const resolveRouting =
   (routing: Routing) =>
   (res: HttpResponse) =>
   (req: HttpRequest): Observable<EffectResponse> => {
-    if (res.finished) {
-      return EMPTY;
-    }
+    if (res.finished) { return EMPTY; }
 
     const [urlPath, urlQuery] = req.url.split('?');
     const routeMatched = findRoute(routing, urlPath, req.method);
 
-    if (!routeMatched) {
-      return EMPTY;
-    }
+    if (!routeMatched) { return EMPTY; }
 
     req.query = queryParamsFactory(urlQuery);
     req.params = routeMatched.params;
 
     const middleware = routeMatched.middleware;
-    const req$ = of(req);
 
     return middleware
-      ? routeMatched.effect(middleware(req$, res, null), res, null)
-      : routeMatched.effect(req$, res, null);
+      ? middleware(of(req), res, null).pipe(
```

Comment:
Obviously this has to be done, but can you check how this affects the performance by running performance tests?

Reply:
Lol, there were many thing added and removed in the meantime 💩It will not tell you and me anything...

Reply:
I'm just curious about the current performance, can you check it?

Reply:
```bash
$ make -C benchmarks

  1 endpoints -- Marble.js
  7843.71

  5 endpoints -- Marble.js
  7374.25

  10 endpoints -- Marble.js
  8547.39

  15 endpoints -- Marble.js
  8319.51

  20 endpoints -- Marble.js
  8010.98

  30 endpoints -- Marble.js
  8425.37

  50 endpoints -- Marble.js
  8388.34

  100 endpoints -- Marble.js
  8053.74
```